tests/unit/detection/test_detection_model.py

```diff
@@ -69,71 +78,133 @@ def test_detection_learner_init_model(od_detection_dataset):
 
 
 @pytest.mark.gpu
-def test_detection_learner_train_one_epoch(od_detection_learner):
+def test_detection_learner_train_one_epoch(
+    od_detection_learner,
+    od_detection_mask_learner
+):
     """ Simply test that a small training loop works. """
     od_detection_learner.fit(epochs=1)
+    # test mask learner
+    od_detection_mask_learner.fit(epochs=1)
 
 
 @pytest.mark.gpu
-def test_detection_learner_plot_precision_loss_curves(od_detection_learner):
+def test_detection_learner_plot_precision_loss_curves(
+    od_detection_learner,
+    od_detection_mask_learner
+):
     """ Simply test that `plot_precision_loss_curves` works. """
     od_detection_learner.plot_precision_loss_curves()
+    # test mask learner
+    od_detection_mask_learner.plot_precision_loss_curves()
 
 
 @pytest.mark.gpu
-def test_detection_learner_evalute(od_detection_learner):
+def test_detection_learner_evaluate(
+    od_detection_learner,
+    od_detection_mask_learner
+):
     """ Simply test that `evaluate` works. """
     od_detection_learner.evaluate()
+    # test mask learner
+    od_detection_mask_learner.evaluate()
 
 
 @pytest.mark.gpu
-def test_detection_learner_predict(od_detection_learner, od_cup_path):
+def test_detection_learner_predict(
+    od_detection_learner,
+    od_cup_path,
+    od_detection_mask_learner
+):
     """ Simply test that `predict` works. """
     bboxes = od_detection_learner.predict(od_cup_path)
     assert type(bboxes) == list
+    # test mask learner
+    bboxes, masks = od_detection_mask_learner.predict(od_cup_path, threshold=0.1)
+    assert type(bboxes) == list
+    assert type(masks) == np.ndarray
+    assert len(bboxes) == len(masks)
 
 
 @pytest.mark.gpu
 def test_detection_learner_predict_threshold(
```

Comment:
this one definitely feels like it *should* be split up between detection and mask